<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Control ESP32 - Germini</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox_Connection">
      <property name="title">
       <string>Conexión Serial</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_Connection">
       <item>
        <widget class="QLabel" name="label_Port">
         <property name="text">
          <string>Puerto:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_Port">
         <property name="minimumSize">
          <size>
           <width>150</width>
           <height>0</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_RefreshPorts">
         <property name="text">
          <string>Actualizar</string>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>16777215</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_Connect">
         <property name="text">
          <string>Conectar</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_Disconnect">
         <property name="text">
          <string>Desconectar</string>
         </property>
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_Status">
         <property name="text">
          <string>Desconectado</string>
         </property>
         <property name="styleSheet">
          <string>color: red; font-weight: bold;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_Control">
      <property name="title">
       <string>Control LED</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_Control">
       <item>
        <widget class="QPushButton" name="pushButton_LED_ON">
         <property name="text">
          <string>Encender LED</string>
         </property>
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="styleSheet">
          <string>background-color: #4CAF50; color: white; font-weight: bold; padding: 10px;</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_LED_OFF">
         <property name="text">
          <string>Apagar LED</string>
         </property>
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="styleSheet">
          <string>background-color: #f44336; color: white; font-weight: bold; padding: 10px;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <widget class="QWidget" name="widget_Uptime" native="true">
       <layout class="QVBoxLayout" name="verticalLayout_Uptime">
        <item>
         <widget class="QLabel" name="label_Uptime">
          <property name="text">
           <string>Tiempo de Actividad:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="textEdit_Uptime">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>60</height>
           </size>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="placeholderText">
           <string>El tiempo de actividad del ESP32 aparecerá aquí...</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget_Debug" native="true">
       <layout class="QVBoxLayout" name="verticalLayout_Debug">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_Debug">
          <item>
           <widget class="QLabel" name="label_Debug">
            <property name="text">
             <string>Comunicación Serial (Debug):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_ClearDebug">
            <property name="text">
             <string>Limpiar</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="textEdit_Debug">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>9</pointsize>
           </font>
          </property>
          <property name="placeholderText">
           <string>Aquí se mostrará toda la comunicación serial...</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>500</width>
     <height>17</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
